### PR TITLE
feat: Drop support for Python 3.9 as EOL reached and add Python 3.14 support

### PR DESCRIPTION
**Pull request by Lee-W**
## Description

Python 3.9 has become upsoorpted version by Oct 31th 
https://devguide.python.org/versions/#unsupported-versions

## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- ~~[ ] Add test cases to all the changes you introduce~~
- [x] Run `poetry all` locally to ensure this change passes linter check and tests
- [x] Manually test the changes:
  - [x] Verify the feature/bug fix works as expected in real-world scenarios
  - [x] Test edge cases and error conditions
  - [x] Ensure backward compatibility is maintained
  - [x] Document any manual testing steps performed
- [x] Update the documentation for the changes

### Documentation Changes

- [ ] Run `poetry doc` locally to ensure the documentation pages renders correctly
- [ ] Check and fix any broken links (internal or external) in the documentation

> When running `poetry doc`, any broken internal documentation links will be reported in the console output like this:
>
> ```text
> INFO    -  Doc file 'config.md' contains a link 'commands/bump.md#-post_bump_hooks', but the doc 'commands/bump.md' does not contain an anchor '#-post_bump_hooks'.
> ```

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,10 +39,10 @@ classifiers = [
     "Programming Language :: Python",
     "Programming Language :: Python :: 3",
     "Programming Language :: Python :: 3.9",
```
Comment (bearomorphism) [inline pyproject.toml]: Wrong deletion?

Comment (Lee-W) [reply pyproject.toml]: 🤦‍♂️

Comment (Lee-W) [reply pyproject.toml]: just updated

**Comment by bearomorphism** [comment]
I think we can merge this first so we can cleanly merge other PRs.

**Comment by Lee-W** [comment]
> I think we can merge this first so we can cleanly merge other PRs.

Yep, sounds like a good idea. @woile could you please take a quick look when you have some bandwidth? Thanks!
